### PR TITLE
Global carrays to stdarray

### PR DESCRIPTION
**Pull request by SyrianBallaS**
Kept changes to a minimum, tested for each affected area. More details are in the last commit.

### CODE REVIEW

**Comment by SyrianBallaS** [comment]
Have to resolve some conflicts.

**Comment by SyrianBallaS** [comment]
Works the same as before, conflicts are resolved. Used KDiff3 making remote whitespace changes preferential to local changes (pulled in RetroAchievements/master to resolve), on to the next one.

**Comment by coczero** [comment]
I'm sorry, I would really like to stay subscribed to this repo, probably even contribute when I find the time, but can you maybe put pull requests with this many commits to your own fork?
I appreciate the work you do for this project but this Email spam is really annoying...

File: src/RA_Leaderboard.cpp [hunk 1]
```diff
@@ -5,20 +5,21 @@
...
+constexpr std::array<const char*, RA_Leaderboard::FormatType::Format__MAX> FormatTypeToString{ 
+    "TIME",      // TimeFrames
+    "TIMESECS",  // TimeSecs
+    "MILLISECS", // TimeMillisecs
+    "SCORE",     // Score /POINTS
+    "VALUE",     //	Value
+    "OTHER",     // Other
 };
-static_assert(SIZEOF_ARRAY(FormatTypeToString) == RA_Leaderboard::Format__MAX, "These must match!");
+
```
[the remark this answers is not in the document]
Comment (SyrianBallaS) [reply src/RA_Leaderboard.cpp]: It actually looks fine when you click edit file on GitHub as well, so sometimes it's out of my hands.

image

**Comment by SyrianBallaS** [comment]
@coczero I understand, I'll keep commits to a minimum.

**Comment by coczero** [comment]
@SyrianBallaS I don't want to discourage you from doing lots of commits (because with big changes you want multiple small commits), but maybe move these things to your personal RA fork and then create a pull request here when your PR is ready :)

**Comment by SyrianBallaS** [comment]
Alright I got you, well I'm gonna leave this one open since I don't think
there's anything else I need to do with it. I had a lot of other ones but
they're closed because they need some work. Functionality-wise they're
ready, but need to be tweaked for the CONTRIBUTING and stuff like that. Are
you getting emails about the closed PRs?

**Comment by coczero** [comment]
Yea I get Mails from closed PRs too, but don't worry. It's ok if there are Emails, I don't mind a few per day. 
I just wanted to say that it makes more sense to put these kind of PRs on your own fork until theyre done, because everyone gets tons of email and for review purposes we need to see the whole finished code anway.

**Comment by SyrianBallaS** [comment]
Alright, I guess I'll just branch out the branch that my pull request is in, commit to that , and merge. I'm not subscribed to this repo so I don't see every message so I'll do that and see if commits from a child branch will reach here. I don't think it will other than the merge commit.

**Comment by SyrianBallaS** [comment]
closing this until the conflicts are fixed

**Pull request closed without merging**